torch/ao/quantization/pt2e/port_metadata_pass.py

```diff
@@ -0,0 +1,215 @@
+import logging
+from typing import Dict, List, Optional, Set
+
+import torch
+from torch._export.error import InternalError
+from torch._export.pass_base import _ExportPassBase
+from torch._ops import HigherOrderOperator, OpOverload, OpOverloadPacket
+
+from torch.ao.quantization.quantizer import QuantizationAnnotation, QuantizationSpecBase
+
+from torch.ao.quantization.quantizer.utils import _is_sym_size_node
+from torch.fx.passes.infra.pass_base import PassResult
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+__all__ = ["PortNodeMetaForQDQ"]
+
+_METADATA_TO_PORT = [
+    "nn_module_stack",
```

Comment:
what about source_fn and val?

Reply:
source_fn is not valid because it also contains nn.Module and I dont want someone to inadvertently grab nodes belonging to say nn.Linear, and now those nodes also include q/dq nodes.

Val is fair one but then we would have to do that as part of convert workflow which should actually quantize the value and dequantize it. wdyth

Reply:
OK, yeah I think it makes sense to not add source_fn here, and should the same reasoning should apply for nn_module_stack as well? e.g. people might be looking for things in nn.Linear module scope, then it will get q/dq ops, will this be surprising? not too familiar with "stack_trace", maybe that one is fine

I thought "val" is storing shape and dtypes, is that true? yeah we do need to update the dtype I think